Sources/GRPC/GRPCChannelHandler.swift

```diff
@@ -68,14 +71,17 @@ extension GRPCChannelHandler: ChannelInboundHandler, RemovableChannelHandler {
     let requestPart = self.unwrapInboundIn(data)
     switch requestPart {
     case .head(let requestHead):
-      guard let callHandler = getCallHandler(channel: context.channel, requestHead: requestHead) else {
-        errorCaught(context: context, error: GRPCError.server(.unimplementedMethod(requestHead.uri)))
+      guard let callHandler = self.makeCallHandler(channel: context.channel, requestHead: requestHead) else {
+        self.errorCaught(context: context, error: GRPCError.server(.unimplementedMethod(requestHead.uri)))
         return
       }
 
+      logger.info("received request head, configuring pipeline")
```

Comment:
> Should we try attaching `, metadata: ["path": "\(requestHead.uri)"]` to all log statements relevant to a specific call?

How about just a uuid so we can reuse the logger across different handlers? 

It may be worth creating a `CallHandlerContext` and passing that to `handleMethod` and then the `*CallHandler`s (instead of the method name, request head, channel, error delegate, and logger) so we can avoid breaking the generated code so often.

> Also, should we have one dedicated canonical "access log" statement that serves a similar purpose as e.g. Apache/nginx access logs? Preferably including request path, response status, and the overall time the request handler took. (Let me know if I missed one you have for that.)

We should!